aten/src/ATen/core/LazyTensor.cpp

```diff
@@ -0,0 +1,142 @@
+#include <ATen/core/LazyTensor.h>
+#include <ATen/core/function_schema.h>
+#include <torch/csrc/autograd/function.h>
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/jit/autodiff.h>
+#include <torch/csrc/jit/graph_executor.h>
+#include <torch/csrc/jit/interpreter.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/script/function_schema_parser.h>
+
+namespace at {
+
+// Embed the LazyTensorImpl into the passed in graph and
+// update map of Value*'s in the graph to their materialized
+// IValues.
+//
+// Returns a Value* in the graph.
+//
+// TODO: Make this non-recursive
+torch::jit::Value* getValue(
+    const LazyTensorImpl* lt,
+    std::shared_ptr<torch::jit::Graph> g,
+    std::map<torch::jit::Value*, c10::IValue>& value_map) {
```

Comment:
Err... a `std::map`? Really?

Reply:
(is this for determinism?)

Reply:
will swap to unordered, oversight on my part